src/App.js

```diff
@@ -13,25 +13,41 @@ const fetchProducts = async () => {
 
   return data;
 };
+const fetchUserData = async (token) => {
+  const data = await callApi({
+    url: "users/me",
+    token,
+  });
+  console.log("user data is:", data);
+  return data;
+};
 
 function App() {
   const [token, setToken] = useState(localStorage.getItem("token"));
   const [products, setProducts] = useState([]);
-  const [userData, setUserData] = useState([]);
+  const [userData, setUserData] = useState({});
   const [cart, setCart] = useState(localStorage.getItem("cart") || {});
 
   useEffect(() => {
     const getProducts = async () => {
       try {
         const products = await fetchProducts();
+        const userData = await fetchUserData(token);
+        const username = userData.username;
```

Comment:
lines 35 and 36 need to be moved to inside the if ( token ) conditional

Reply:
I think we could have them all on the same page, the my-orders button  was more of a place holder for the time being since I saw you where working on an orders component.